### PR TITLE
allow passing `headers` to subgraphs

### PR DESCRIPTION
**Pull request by mikaelkaron**
see https://github.com/platformatic/platformatic/issues/1966

### CODE REVIEW

File: lib/composer.js [hunk 1]
```diff
@@ -462,7 +462,7 @@ class Composer {
       info
     })
 
-    const result = await this.runQueries(queries, order)
+    const result = await this.runQueries(queries, order, excludeKeys(context.reply.request.headers, ['content-type', 'content-length']))
```
Comment (mikaelkaron) [inline lib/composer.js:465]: Is there a better way of getting a hold of the request headers? (one may have thought `context.request.headers` would have worked)

File: lib/composer.js [hunk 2]
```diff
@@ -462,7 +462,7 @@ class Composer {
       info
     })
 
-    const result = await this.runQueries(queries, order)
+    const result = await this.runQueries(queries, order, excludeObjectKeys(context.reply.request.headers, ['content-type', 'content-length']))
```
Comment (mikaelkaron) [inline lib/composer.js:465]: Is there a better way of getting a hold of the request headers? (one could have thought `context.request.headers` would have worked)

Comment (mcollina) [reply lib/composer.js:465]: That’s the correct way to do it unfortunately

**Review by mcollina — COMMENTED** [review]
Can you add a test?

**Comment by mikaelkaron** [comment]
> Can you add a test?

Of course. That’s why it’s a WIP ;)

**Comment by mikaelkaron** [comment]
A few notes on the implementation:

1. Is it enough to just have a blacklist of headers to remove
2. Should there also be a "whitelist" of headers to keep
3. Is there a need to transform headers
4. Should there be a way to inject whitelist/blacklist/transformer from the outside

So I'm not sure about 1, I'm _hoping_ that it's OK to just forward all headers (except a few) but I can already think about problems this may cause downstream.

The problem with 1 can be partially solved with 2, but it adds complexity that may not be needed at this stage.

I _don't_ think we should transform headers _here_ - if it's needed it should happen _before_ we get here.

Providing some configuration would probably be a good idea - I'll try to add this without adding complexity. I'm hoping this can be simple enough to be serialized and provided in a platformatic configuration in the end.

**Comment by mcollina** [comment]
IMHO having a callback to transform the headers would solve all the configurability problems. _However_, keep this implementation as simple as possible, and we can add that later on if needed.

**Comment by mcollina** [comment]
Any help with the tests?

**Comment by mikaelkaron** [comment]
Sorry. I’ll try to get to it today.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mikaelkaron** [comment]
Sorry @mcollina - just pushed a change renaming the test to align with other test names

**Pull request merged**